pypeit/core/telluric.py

```diff
@@ -1718,199 +1764,377 @@ def poly_telluric(spec1dfile, telgridfile, telloutfile, outfile, z_obj=0.0, func
+            details.
+        popsize (:obj:`int`, optional):
+            A multiplier for setting the total population size for the
+            differential evolution optimization. See
+            `scipy.optimize.differential_evolution`_ for details.
+        recombination (:obj:`float`, optional):
+            The recombination constant for the differential evolution
+            optimization. This should be in the range [0, 1]. See
+            `scipy.optimize.differential_evolution`_ for details.
+        polish (:obj:`bool`, optional):
+            If True then differential evolution will perform an additional
+            optimization at the end to polish the best fit at the end, which
+            can improve the optimization slightly. See
+            `scipy.optimize.differential_evolution`_ for details.
+        disp (:obj:`bool`, optional):
+            Argument for `scipy.optimize.differential_evolution`_ that prints
+            status messages to the screen indicating the status of the
+            optimization. See description above.
+        sensfunc (:obj:`bool`, optional):
+            This option is used for usage of this class for joint telluric
+            fitting and sensitivity function computation. If True, the input
+            flux is in counts and then converted to counts per angstrom,
+            since the sensfunc is obtained by fitting counts per angstrom.
+            TODO: This explanation is unclear.
+        debug (:obj:`bool`, optional):
+            If True, QA plots will be shown to the screen indicating the
+            quality of the fits. Specifically, the residual distributions
+            will be shown at each iteration, and the fit will be shown at the
+            end (for each order). This is useful if you are running the code
+            for the first time, but since the algorithm is slow, particularly
+            for fitting multi-order echelle data, it will require lots of
+            clicking to close interactive matplotlib windows which halt code
+            flow.
+
     """
-    def __init__(self, wave, flux, ivar, mask, telgridfile, obj_params, init_obj_model, eval_obj_model,
-                 ech_orders=None,
-                 sn_clip=30.0, airmass_guess=1.5, resln_guess=None,
-                 resln_frac_bounds=(0.5, 1.5), pix_shift_bounds=(-5.0, 5.0), pix_stretch_bounds=(0.9,1.1),
-                 maxiter=2, sticky=True, lower=3.0, upper=3.0,
-                 seed=777, ballsize = 5e-4, tol=1e-3, diff_evol_maxiter=1000,  popsize=30, recombination=0.7, polish=True, disp=False, sensfunc=False,
-                 debug=False):
+    version = '1.0.0'
+    """Datamodel version."""
+
+    datamodel = {'telgrid': dict(otype=str,
+                                 descr='File containing grid of HITRAN atmosphere models'),
+                 'std_src': dict(otype=str, descr='Name of the standard source'),
+                 'std_name': dict(otype=str, descr='Type of standard source'),
```

Comment:
Can you list the possible options here?  If they are only a few.

Reply:
This is a question for @feigewang and @jhennawi .